### PR TITLE
Order Views and CRUD,reused functions,roles type changes for User

### PR DESCRIPTION
**Pull request by nikhild007**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -4,9 +4,10 @@ class ApplicationController < ActionController::Base
...
   private
-    def user_not_authorized
-      @error = "User not authorized to perform this action"
-      flash[:error] = @error
-      redirect_to controller: "products", action: "new"
+    def user_not_authorized(exception)
+      policy_name = exception.policy.class.to_s.underscore
+      dynamic_controller = policy_name.split("_")[0]
+      flash[:error] = "#{policy_name}.#{exception.query} not allowed for your role"
+      redirect_to controller: 'home', action: 'index'
```
Comment (RAJASHARMA) [inline app/controllers/application_controller.rb]: You can replace it something to go back to previous page

redirect_back(fallback_location: root_path)

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
-            render 'cart/index', status: :unprocessable_entity
+            flash[:error] = "Order Failed"
+            redirect_to controller: 'cart', action: 'index'
+        end
+    end
+
+    def destroy
+        authorize @order, :destroy?
+        if @order.destroy!
+            redirect_to controller:"home", action: "index"
```
Comment (RAJASHARMA) [inline app/controllers/orders_controller.rb:34]: Any success message. ?

File: app/helpers/orders_helper.rb [hunk 3]
```diff
@@ -1,13 +1,38 @@
 module OrdersHelper
+    ORDER_STATUS_MESSAGES = {
+       "accepted" => "Order Accepted",
+       "dispatched" => "Dispatched",
+       "cancel" => "Cancelled",
```
Comment (RAJASHARMA) [inline app/helpers/orders_helper.rb]: I believe it should be "cancelled"

File: app/controllers/orders_controller.rb (hunk 2 above)
Comment (nikhild007) [reply app/controllers/orders_controller.rb:34]: Yaa that part is not done yet as this for admin side ,i will add a success message

**Pull request merged**